pkg/promclient/promclient.go

```diff
@@ -181,26 +178,42 @@ func ConfiguredFlags(ctx context.Context, logger log.Logger, base *url.URL) (Fla
 	if err != nil {
 		return Flags{}, errors.Wrap(err, "create request")
 	}
-	resp, err := http.DefaultClient.Do(req.WithContext(ctx))
-	if err != nil {
-		return Flags{}, errors.Wrapf(err, "request config against %s", u.String())
-	}
-	defer runutil.CloseWithLogOnErr(logger, resp.Body, "query body")
 
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return Flags{}, errors.New("failed to read body")
-	}
+	var body []byte
 
-	if resp.StatusCode != 200 {
-		return Flags{}, errors.Errorf("got non-200 response code: %v, response: %v", resp.StatusCode, string(b))
+	if err := runutil.Retry(2*time.Second, ctx.Done(), func() error {
+
+		resp, err := http.DefaultClient.Do(req.WithContext(ctx))
+		if err != nil {
+			return errors.Wrapf(err, "request config against %s", u.String())
+		}
+		defer runutil.CloseWithLogOnErr(logger, resp.Body, "query body")
+
+		body, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return errors.New("failed to read body")
+		}
+
+		if resp.StatusCode == 200 || resp.StatusCode == 404 { // 404, just give up
+			return nil
```

Comment:
I would say something like
```
var NotFoundFlags = errors.New("No flag endpoint found")
```
has to be returned here. And let caller decide what to do -> e.g we should log on 404 and, grab flags on 200 and error  on anything else IMO (:

Reply:
Sorry for being strict ):

Reply:
But I'm not sure how you want to do that with the retry logic? If I understand that code correctly it retries upon seeing an error, meaning it will retry `NotFoundFlags`?

Reply:
Well we need to pass it somehow. This question is just implementation detail. We either move retry logic to caller (I prefer that - more explicit) and just retry on certain errors or just pass error in separate var here.

Reply:
Ok, moved and added a few things back to not make too many changes.